<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>danrerlib.GO &mdash; danrerlib</title>
      <link rel="stylesheet" href="../../../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../../../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../../../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" type="text/css" />
      <link rel="stylesheet" href="../../../_static/graphviz.css" type="text/css" />
      <link rel="stylesheet" href="../../../_static/custom.css" type="text/css" />
      <link rel="stylesheet" href="../../../_static/fonts.css" type="text/css" />
      <link rel="stylesheet" href="../../../_static/css/custom.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../../../" id="documentation_options" src="../../../_static/documentation_options.js"></script>
        <script src="../../../_static/jquery.js"></script>
        <script src="../../../_static/underscore.js"></script>
        <script src="../../../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../../../_static/doctools.js"></script>
        <script src="../../../_static/sphinx_highlight.js"></script>
    <script src="../../../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../../../genindex.html" />
    <link rel="search" title="Search" href="../../../search.html" />
    <link rel="next" title="danrerlib.KEGG" href="../KEGG/index.html" />
    <link rel="prev" title="danrerlib" href="../index.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../../../index.html" class="icon icon-home">
            danrerlib
              <img src="../../../_static/danrerlib_logo.png" class="logo" alt="Logo"/>
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">User's Guide</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../../userguide/introduction.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../userguide/installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../userguide/usage.html">Usage Principles</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../index.html">API Reference</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">danrerlib</span></code></a><ul class="current">
<li class="toctree-l3 current"><a class="reference internal" href="../index.html#submodules">Submodules</a><ul class="current">
<li class="toctree-l4 current"><a class="current reference internal" href="#"><code class="xref py py-mod docutils literal notranslate"><span class="pre">danrerlib.GO</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../KEGG/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">danrerlib.KEGG</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../enrichment/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">danrerlib.enrichment</span></code></a></li>
<li class="toctree-l4"><a class="reference internal" href="../mapping/index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">danrerlib.mapping</span></code></a></li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Tutorials</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorials/perform_geneid_mapping.html">Gene ID Mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../tutorials/perform_orthology_mapping.html">Orthology Mapping</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Developer's Guide</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../../devguide/changelog.html">Changelog</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../devguide/contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../devguide/conduct.html">Code of Conduct</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../../devguide/license.html">License</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../../index.html">danrerlib</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../../../index.html" class="icon icon-home" aria-label="Home"></a></li>
          <li class="breadcrumb-item"><a href="../../index.html">API Reference</a></li>
          <li class="breadcrumb-item"><a href="../index.html"><code class="xref py py-mod docutils literal notranslate"><span class="pre">danrerlib</span></code></a></li>
      <li class="breadcrumb-item active"><code class="xref py py-mod docutils literal notranslate"><span class="pre">danrerlib.GO</span></code></li>
      <li class="wy-breadcrumbs-aside">
            <a href="../../../_sources/autoapi/danrerlib/GO/index.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="module-danrerlib.GO">
<span id="danrerlib-go"></span><h1><a class="reference internal" href="#module-danrerlib.GO" title="danrerlib.GO"><code class="xref py py-mod docutils literal notranslate"><span class="pre">danrerlib.GO</span></code></a><a class="headerlink" href="#module-danrerlib.GO" title="Permalink to this heading"></a></h1>
<section id="go-module">
<p>The Gene Ontology (GO) module provides functions for retrieving gene information associated with
Gene Ontology Biological Process (BP), Molecularfunctionsns (MF) and Cellular Components (CC)
for various organisms, including human, zebrafish, and mapped zebrafish.</p>
<dl class="simple">
<dt>Functions:</dt><dd><ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">get_genes_in_GO_concept</span></code>: Retrieve genes associated with a specific GO concept ID.</p></li>
</ul>
</dd>
<dt>Constants:</dt><dd><ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">NCBI_ID</span></code>: Identifier for NCBI Gene ID.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">ZFIN_ID</span></code>: Identifier for ZFIN ID.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">ENS_ID</span></code>: Identifier for Ensembl ID.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">SYMBOL</span></code>: Identifier for gene Symbol.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">HUMAN_ID</span></code>: Identifier for Human NCBI Gene ID.</p></li>
</ul>
</dd>
<dt>Database Rebuild Functions:</dt><dd><ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">build_gene_ontology_database</span></code>: Build or update the gene ontology database.</p></li>
</ul>
</dd>
<dt>Notes:</dt><dd><ul class="simple">
<li><p>This module is designed for accessing gene information from Gene Ontology.</p></li>
<li><p>It provides functions to retrieve gene data associated with biological processes, cellular components, and molecular functions.</p></li>
<li><p>Data is obtained either through reading pre-processed files from the database build.</p></li>
<li><p>Organism options include ‘dre’ (Danio rerio), ‘hsa’ (Human), or ‘dreM’ (Mapped Danio rerio from Human).</p></li>
</ul>
</dd>
<dt>Example:</dt><dd><p>To retrieve genes associated with a specific GO concept ID:
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">go_genes</span> <span class="pre">=</span> <span class="pre">get_genes_in_GO_concept('GO:0007582',</span> <span class="pre">'dre')</span>
<span class="pre">`</span></code></p>
</dd>
</dl>
<p>For detailed information on each function and their usage, please refer to the documentation. For more examples of full functionality, please refer to tutorials.</p>
</section>
<section id="module-contents">
<h2>Module Contents<a class="headerlink" href="#module-contents" title="Permalink to this heading"></a></h2>
<section id="functions">
<h3>Functions<a class="headerlink" href="#functions" title="Permalink to this heading"></a></h3>
<table class="autosummary longtable docutils align-default">
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#danrerlib.GO.get_genes_in_GO_concept" title="danrerlib.GO.get_genes_in_GO_concept"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_genes_in_GO_concept</span></code></a>(→ pandas.Series)</p></td>
<td><p>Retrieve gene IDs associated with a Gene Ontology (GO) concept for a specified organism.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#danrerlib.GO.id_exists_given_organism" title="danrerlib.GO.id_exists_given_organism"><code class="xref py py-obj docutils literal notranslate"><span class="pre">id_exists_given_organism</span></code></a>(concept_id, organism)</p></td>
<td><p>Check if a Gene Ontology (GO) ID exists for the specified organism.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#danrerlib.GO.build_gene_ontology_database" title="danrerlib.GO.build_gene_ontology_database"><code class="xref py py-obj docutils literal notranslate"><span class="pre">build_gene_ontology_database</span></code></a>()</p></td>
<td><p>Build or update the Gene Ontology (GO) database.</p></td>
</tr>
</tbody>
</table>
<dl class="py function">
<dt class="sig sig-object py" id="danrerlib.GO.get_genes_in_GO_concept">
<span class="sig-prename descclassname"><span class="pre">danrerlib.GO.</span></span><span class="sig-name descname"><span class="pre">get_genes_in_GO_concept</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">concept_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">organism</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">gene_id_type</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">Optional</span><span class="p"><span class="pre">[</span></span><span class="pre">str</span><span class="p"><span class="pre">]</span></span></span><span class="w"> </span><span class="o"><span class="pre">=</span></span><span class="w"> </span><span class="default_value"><span class="pre">None</span></span></em><span class="sig-paren">)</span> <span class="sig-return"><span class="sig-return-icon">&#x2192;</span> <span class="sig-return-typehint"><span class="pre">pandas.Series</span></span></span><a class="headerlink" href="#danrerlib.GO.get_genes_in_GO_concept" title="Permalink to this definition"></a></dt>
<dd><p>Retrieve gene IDs associated with a Gene Ontology (GO) concept for a specified organism.</p>
<dl>
<dt>Parameters:</dt><dd><ul>
<li><p><code class="docutils literal notranslate"><span class="pre">concept_id</span> <span class="pre">(str)</span></code>: The Gene Ontology (GO) concept ID for which gene IDs are to be retrieved.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">organism</span> <span class="pre">(str)</span></code>: The organism for which gene IDs should be retrieved. Options include</p>
<blockquote>
<div><ul class="simple">
<li><p>‘hsa’: Human.</p></li>
<li><p>‘dre’: Zebrafish.</p></li>
<li><p>‘dreM’: Mapped Zebrafish from Human (same as ‘dre’ as Zebrafish GO IDs are not characterized).</p></li>
</ul>
</div></blockquote>
</li>
<li><p><code class="docutils literal notranslate"><span class="pre">gene_id_type</span> <span class="pre">(str,</span> <span class="pre">optional)</span></code>: The desired gene ID type for the returned gene IDs. Default is None.</p></li>
</ul>
</dd>
<dt>Returns:</dt><dd><ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">gene_ids_series</span> <span class="pre">(pd.Series)</span></code>: A pandas Series containing gene IDs associated with the specified GO concept.</p></li>
</ul>
</dd>
<dt>Raises:</dt><dd><p>ValueError: If the GO concept ID does not exist for the given organism or if the organism is invalid.</p>
</dd>
<dt>Notes:</dt><dd><ul class="simple">
<li><p>This function retrieves gene IDs associated with a Gene Ontology (GO) concept for a specified organism.</p></li>
<li><p>The ‘concept_id’ parameter should be a valid GO concept identifier.</p></li>
<li><p>The ‘organism’ parameter specifies the organism for which gene IDs should be retrieved.</p></li>
<li><p>Options for ‘organism’ include ‘hsa’ (Human), ‘dre’ (Zebrafish), or ‘dreM’ (Mapped Zebrafish from Human).</p></li>
</ul>
</dd>
<dt>Example:</dt><dd><p>To retrieve Zebrafish gene IDs associated with a GO concept:
<code class="docutils literal notranslate"><span class="pre">`</span>
<span class="pre">concept_id</span> <span class="pre">=</span> <span class="pre">'GO:0001234'</span>
<span class="pre">organism</span> <span class="pre">=</span> <span class="pre">'dre'</span>
<span class="pre">gene_ids</span> <span class="pre">=</span> <span class="pre">get_genes_in_GO_concept(concept_id,</span> <span class="pre">organism)</span>
<span class="pre">`</span></code></p>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="danrerlib.GO.id_exists_given_organism">
<span class="sig-prename descclassname"><span class="pre">danrerlib.GO.</span></span><span class="sig-name descname"><span class="pre">id_exists_given_organism</span></span><span class="sig-paren">(</span><em class="sig-param"><span class="n"><span class="pre">concept_id</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em>, <em class="sig-param"><span class="n"><span class="pre">organism</span></span><span class="p"><span class="pre">:</span></span><span class="w"> </span><span class="n"><span class="pre">str</span></span></em><span class="sig-paren">)</span><a class="headerlink" href="#danrerlib.GO.id_exists_given_organism" title="Permalink to this definition"></a></dt>
<dd><p>Check if a Gene Ontology (GO) ID exists for the specified organism.</p>
<dl class="simple">
<dt>Parameters:</dt><dd><ul class="simple">
<li><p><code class="docutils literal notranslate"><span class="pre">concept_id</span> <span class="pre">(str)</span></code>: The GO ID to be validated.</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">organism</span> <span class="pre">(str)</span></code>: The specified organism code (e.g., ‘hsa’, ‘dre’, or ‘dreM’).</p></li>
</ul>
</dd>
<dt>Returns:</dt><dd><ul class="simple">
<li><p>bool: True if the GO ID exists for the specified organism, False otherwise.</p></li>
</ul>
</dd>
<dt>Raises:</dt><dd><ul class="simple">
<li><p>ValueError: If the organism code is invalid.</p></li>
</ul>
</dd>
<dt>Notes:</dt><dd><ul class="simple">
<li><p>This function checks whether a given GO ID exists for the specified organism.</p></li>
<li><p>It reads a GO ID data file and looks for a match in the ‘GO ID’ column.</p></li>
<li><p>The ‘organism’ parameter specifies the organism code for which to perform the check.</p></li>
<li><p>Returns True if the GO ID exists for the organism, False otherwise.</p></li>
<li><p>Provides an error message if the organism code is invalid.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py function">
<dt class="sig sig-object py" id="danrerlib.GO.build_gene_ontology_database">
<span class="sig-prename descclassname"><span class="pre">danrerlib.GO.</span></span><span class="sig-name descname"><span class="pre">build_gene_ontology_database</span></span><span class="sig-paren">(</span><span class="sig-paren">)</span><a class="headerlink" href="#danrerlib.GO.build_gene_ontology_database" title="Permalink to this definition"></a></dt>
<dd><p>Build or update the Gene Ontology (GO) database.</p>
<dl>
<dt>Notes:</dt><dd><ul>
<li><p>This function is intended for database creation or update and should be run only during version updates.</p></li>
<li><p>The process may take some time, so exercise caution when running it.</p></li>
<li><p>The function performs the following steps:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>Downloads all current GO IDs.</p></li>
<li><p>Builds the human GO file and with IDs and associated genes.</p></li>
</ol>
<p>3. Builds the zebrafish GO file and with IDs and associated genes.
5. Builds the mapped zebrafish GO file and with IDs and associated genes.
6. Add a column to the database file to identify if the GO ID exists for human and/or zebrafish.</p>
</div></blockquote>
</li>
<li><p>Running this function should be done carefully, as it involves downloading and processing data fromGO.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</section>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="../index.html" class="btn btn-neutral float-left" title="danrerlib" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="../KEGG/index.html" class="btn btn-neutral float-right" title="danrerlib.KEGG" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Ashley Schwartz.</p>
  </div>

   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>