Search.setIndex({"docnames": ["_templates/autoapi/index", "autoapi/danrerlib/GO/index", "autoapi/danrerlib/KEGG/index", "autoapi/danrerlib/database/index", "autoapi/danrerlib/enrichment/index", "autoapi/danrerlib/index", "autoapi/danrerlib/mapping/index", "autoapi/danrerlib/settings/index", "autoapi/danrerlib/utils/index", "autoapi/index", "devguide/changelog", "devguide/conduct", "devguide/contributing", "devguide/license", "index", "tutorials/index", "tutorials/perform_geneid_mapping", "tutorials/perform_orthology_mapping", "userguide/installation", "userguide/introduction", "userguide/usage"], "filenames": ["_templates/autoapi/index.rst", "autoapi/danrerlib/GO/index.rst", "autoapi/danrerlib/KEGG/index.rst", "autoapi/danrerlib/database/index.rst", "autoapi/danrerlib/enrichment/index.rst", "autoapi/danrerlib/index.rst", "autoapi/danrerlib/mapping/index.rst", "autoapi/danrerlib/settings/index.rst", "autoapi/danrerlib/utils/index.rst", "autoapi/index.rst", "devguide/changelog.md", "devguide/conduct.md", "devguide/contributing.md", "devguide/license.md", "index.md", "tutorials/index.md", "tutorials/perform_geneid_mapping.ipynb", "tutorials/perform_orthology_mapping.ipynb", "userguide/installation.md", "userguide/introduction.md", "userguide/usage.md"], "titles": ["API Reference", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">danrerlib.GO</span></code>", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">danrerlib.KEGG</span></code>", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">danrerlib.database</span></code>", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">danrerlib.enrichment</span></code>", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">danrerlib</span></code>", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">danrerlib.mapping</span></code>", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">danrerlib.settings</span></code>", "<code class=\"xref py py-mod docutils literal notranslate\"><span class=\"pre\">danrerlib.utils</span></code>", "API Reference", "Changelog", "Code of Conduct", "Contributing", "License", "Welcome to danRerLib\u2019s Documentation", "Tutorials", "Gene ID Mapping", "Orthology Mapping", "Installation", "Introduction", "Usage Principles"], "terms": {"thi": [0, 1, 2, 4, 6, 8, 9, 11, 12, 13, 16, 17, 18, 19, 20], "page": [0, 9, 14, 20], "contain": [0, 1, 2, 4, 6, 9, 16, 17], "document": [0, 1, 2, 4, 6, 9, 13, 16, 19], "The": [1, 2, 4, 6, 8, 11, 12, 13, 16, 17, 18, 20], "gene": [1, 2, 4, 6, 7, 8, 14, 15, 19], "ontologi": [1, 4, 19, 20], "provid": [1, 2, 4, 6, 8, 13, 19], "retriev": [1, 2, 4, 6, 16, 19, 20], "inform": [1, 2, 4, 6, 11, 14, 16, 17, 18, 19, 20], "associ": [1, 2, 4, 13, 20], "biolog": [1, 4, 19, 20], "process": [1, 2, 4, 6, 19, 20], "bp": [1, 4, 19, 20], "molecularfunctionsn": 1, "mf": [1, 4, 19, 20], "cellular": [1, 4, 19, 20], "compon": [1, 4, 19, 20], "cc": [1, 4, 19, 20], "variou": [1, 2, 4, 17], "organ": [1, 2, 4, 6, 8, 16, 17], "includ": [1, 2, 4, 6, 8, 11, 12, 13, 16, 17], "human": [1, 2, 4, 6, 7, 19], "zebrafish": [1, 2, 4, 6, 8, 14, 16, 19], "map": [1, 2, 4, 5, 9, 14, 15, 19, 20], "get_genes_in_go_concept": 1, "specif": [1, 2, 11, 14, 16, 18, 19, 20], "concept": [1, 4, 19], "id": [1, 2, 4, 6, 7, 8, 14, 15, 17, 19], "constant": [1, 2, 4, 6], "ncbi_id": [1, 2, 4, 6, 7, 20], "identifi": [1, 2, 4, 6, 19, 20], "ncbi": [1, 2, 4, 6, 7, 8, 16, 17, 19, 20], "zfin_id": [1, 2, 4, 6, 7, 20], "zfin": [1, 2, 4, 6, 7, 8, 16, 17, 19, 20], "ens_id": [1, 2, 4, 6, 7, 20], "ensembl": [1, 2, 4, 6, 7, 8, 16, 17, 19, 20], "symbol": [1, 2, 4, 6, 7, 8, 16, 17, 19, 20], "human_id": [1, 2, 4, 6, 7, 17], "databas": [1, 2, 4, 6, 16, 17, 19], "rebuild": [1, 2, 6, 19], "build_gene_ontology_databas": 1, "build": [1, 2, 6, 16, 17, 19], "updat": [1, 2, 12, 14, 19], "note": [1, 2, 4, 6, 8, 12, 16, 17], "i": [1, 2, 4, 6, 8, 11, 12, 13, 14, 16, 17, 18, 19, 20], "design": [1, 2, 4, 6, 14, 19], "access": [1, 2], "from": [1, 2, 4, 6, 11, 12, 13, 19, 20], "It": [1, 2, 4, 6, 16, 17, 18, 19, 20], "data": [1, 2, 6, 8, 16, 17, 19], "molecular": [1, 4, 19, 20], "obtain": [1, 2, 13], "either": [1, 2, 16, 17, 18], "through": [1, 2, 12, 16, 17], "read": [1, 2, 6, 16, 17], "pre": [1, 2], "file": [1, 2, 6, 13], "option": [1, 2, 4, 6, 8, 16, 17], "dre": [1, 2, 4, 8], "danio": [1, 2, 14], "rerio": [1, 2, 14], "hsa": [1, 2, 4, 8], "drem": [1, 2, 4, 8], "exampl": [1, 2, 4, 6, 11, 16, 17, 20], "To": [1, 2, 4, 6, 16, 17, 20], "go_gen": 1, "0007582": [1, 4], "For": [1, 2, 4, 6, 14, 16, 17], "detail": [1, 2, 4, 6, 11, 12, 14, 19], "each": [1, 2, 4, 6, 16, 20], "usag": [1, 2, 4, 6, 14, 19], "pleas": [1, 2, 4, 6, 12, 16, 18, 19, 20], "refer": [1, 2, 6, 16, 18, 19, 20], "more": [1, 2, 6, 8, 14, 16, 17, 19, 20], "full": [1, 2, 4, 6, 14, 16], "tutori": [1, 2, 4, 6, 16, 17, 19, 20], "concept_id": [1, 4], "str": [1, 2, 4, 6, 8, 20], "gene_id_typ": [1, 4, 8], "none": [1, 2, 4, 6, 8], "panda": [1, 2, 6, 16, 17, 18], "seri": [1, 6, 8], "specifi": [1, 2, 4, 6, 8, 16, 17], "paramet": [1, 2, 4, 6, 8, 16], "which": [1, 2, 11, 16, 17], "ar": [1, 2, 6, 8, 11, 12, 16, 17, 18, 20], "should": [1, 2, 12, 16, 17, 18], "same": [1, 2, 16, 17], "character": [1, 2, 17], "desir": [1, 6], "type": [1, 4, 6, 8, 16, 17, 19], "return": [1, 2, 4, 6, 8, 16], "default": [1, 2, 4, 6, 16, 17], "gene_ids_seri": 1, "pd": [1, 2, 4, 6, 8, 16, 17], "A": [1, 2, 4, 6, 8, 13, 14, 16, 17, 20], "rais": [1, 6, 8], "valueerror": 1, "If": [1, 2, 6, 8, 12, 14, 16, 17, 18, 19], "doe": [1, 16, 17], "exist": [1, 16, 17], "given": [1, 4, 6, 12, 16, 17], "invalid": [1, 6, 8], "valid": [1, 2, 8], "0001234": 1, "gene_id": 1, "id_exists_given_organ": 1, "check": [1, 6, 8, 12, 14, 16], "code": [1, 4, 14, 16, 17], "e": [1, 4, 11, 17], "g": [1, 4, 17], "bool": [1, 2, 4, 6], "true": [1, 2, 4, 6, 16, 17], "fals": [1, 2, 4, 6, 16, 17], "otherwis": [1, 6, 11, 13, 16, 17], "whether": [1, 2, 4, 6, 13], "look": [1, 12, 16, 17], "match": [1, 6, 16, 17], "column": [1, 6], "perform": [1, 2, 4, 6, 19], "an": [1, 8, 11, 13, 16, 17, 18], "error": 1, "messag": 1, "intend": [1, 2], "creation": [1, 2], "run": [1, 2, 6, 16, 17, 19], "onli": [1, 2, 4, 16, 17], "dure": [1, 2], "version": [1, 2, 11, 12], "mai": [1, 2, 11, 16], "take": [1, 2, 11, 17], "some": [1, 2, 6, 16, 17, 18, 20], "time": [1, 2, 16], "so": [1, 2, 13], "exercis": [1, 2], "caution": [1, 2], "when": [1, 2, 4, 6, 8, 11, 12, 16, 17], "follow": [1, 2, 11, 13, 16, 17, 18, 19, 20], "step": [1, 2, 12, 14, 16, 17, 18], "download": [1, 2, 12, 18], "all": [1, 11, 12, 13, 16, 17, 18], "current": [1, 6, 12, 16, 17, 18, 20], "3": [1, 16, 17, 18], "5": [1, 16, 17], "6": [1, 16, 17], "add": [1, 6, 12, 16, 17], "done": [1, 2, 12, 16, 17], "carefulli": [1, 2], "involv": [1, 2], "fromgo": 1, "pathwai": [2, 4, 16, 17, 19], "diseas": [2, 4, 20], "get_genes_in_pathwai": 2, "get_genes_in_diseas": 2, "build_kegg_databas": 2, "direct": [2, 4], "api": [2, 16, 19, 20], "call": [2, 16, 17], "pathway_gen": 2, "hsa04010": 2, "pathway_id": 2, "org": [2, 4, 8], "interest": [2, 4, 11, 14, 17, 19], "us": [2, 4, 6, 11, 12, 13, 14, 16, 17, 18, 19, 20], "df": [2, 4], "datafram": [2, 4, 6, 8, 16, 17], "disease_id": 2, "result": [2, 4, 6, 17], "rest": 2, "set": [2, 4, 6, 8, 11, 12, 16, 17, 19, 20], "fetch": 2, "store": [2, 16], "them": [2, 16, 17], "creat": [2, 6, 11, 12, 18], "directori": [2, 6, 16, 17], "offer": 4, "collect": [4, 20], "analys": [4, 19], "These": [4, 17], "allow": 4, "you": [4, 6, 12, 14, 16, 17, 18, 19, 20], "overrepres": [4, 20], "compar": [4, 20], "background": [4, 15], "univers": 4, "support": [4, 6, 12, 16, 17, 20], "method": [4, 16, 17, 20], "make": [4, 6, 11, 12, 14, 18], "versatil": 4, "tool": [4, 18], "uncov": 4, "insight": 4, "enrich_go": 4, "go": [4, 5, 9, 16, 17, 19, 20], "analysi": [4, 17, 19, 20], "enrich_kegg": 4, "conduct": [4, 14, 19], "kyoto": [4, 19], "encyclopedia": [4, 19], "genom": [4, 6, 14, 16, 17, 19], "kegg": [4, 5, 9, 17, 19, 20], "logist": [4, 20], "regress": [4, 20], "fisher": [4, 20], "": [4, 11, 12, 18, 20], "exact": [4, 20], "test": [4, 12, 16, 17, 19, 20], "analyz": 4, "context": 4, "user": 4, "can": [4, 6, 8, 12, 16, 17, 18, 19, 20], "choos": [4, 17], "differ": [4, 6, 11, 16, 17, 19], "enhanc": [4, 12, 19], "flexibl": 4, "statist": [4, 19, 20], "gene_univers": 4, "comprehens": [4, 14, 19], "consult": 4, "also": [4, 16, 17, 20], "find": [4, 17], "demonstr": 4, "gene_set": 4, "concept_typ": 4, "concept_nam": 4, "sig_cutoff": 4, "0": [4, 16, 17, 18], "05": 4, "repres": [4, 8, 11], "term": [4, 12], "being": 4, "name": [4, 6, 8, 12, 16, 17], "descript": [4, 16, 17, 20], "float": 4, "signific": [4, 19], "cutoff": 4, "inclus": [4, 11], "number": [4, 17], "belong": 4, "proport": 4, "coeffici": 4, "p": 4, "valu": [4, 6, 17], "discoveri": 4, "rate": 4, "fdr": 4, "adjust": 4, "odd": 4, "ratio": 4, "statu": 4, "deplet": 4, "calcul": 4, "larger": 4, "base": [4, 6, 8, 16, 18, 20], "slope": 4, "sig_gene_cutoff_pvalu": 4, "sig_conceptid_cutoff_pvalu": 4, "sig_conceptid_cutoff_fdr": 4, "order_by_p_valu": 4, "log2fc": [4, 16, 17], "list": [4, 6, 8, 17, 20], "order": [4, 6, 16, 17], "two": [4, 16, 17], "filter": 4, "applic": 4, "determin": [4, 11, 17, 20], "enrich": [5, 9, 19], "orthologi": [6, 14, 15, 19, 20], "between": [6, 16, 17, 19, 20], "speci": 6, "focu": 6, "convert_id": [6, 16], "convert": [6, 17, 19], "add_mapped_column": [6, 16], "new": [6, 14, 16, 18, 19], "convert_to_human": [6, 17], "correspond": 6, "ortholog": [6, 19], "convert_to_zebrafish": [6, 17], "add_mapped_ortholog_column": [6, 17], "build_gene_map": 6, "master": [6, 16, 17], "convers": [6, 16, 17, 19, 20], "build_ortho_map": 6, "research": [6, 14, 17, 19], "project": [6, 11, 12, 18, 20], "would": [6, 12, 16, 17, 18, 20], "like": [6, 16, 17, 18], "re": [6, 12, 14, 18, 19], "ensur": [6, 19], "your": [6, 12, 16, 17, 18, 19, 20], "dataset": 6, "accur": 6, "per": 6, "requir": [6, 12, 15, 18], "recommend": [6, 19], "packag": [6, 16, 17, 18, 19], "gene_list": 6, "ensdarg00000012345": 6, "ensdarg00000067890": 6, "converted_gen": 6, "except": [6, 8, 17], "databasenotfounderror": 6, "found": [6, 20], "invalidgenetypeerror": [6, 8], "union": [6, 8], "numpi": [6, 18], "arrai": 6, "id_from": [6, 16, 17], "id_to": [6, 16, 17], "keep_map": [6, 16, 17], "out_format": 6, "format": [6, 8, 12, 16, 17], "np": 6, "keep": [6, 12], "origin": [6, 16], "output": [6, 16, 17], "mapped_gen": 6, "other": [6, 11, 13, 16, 17, 18], "import": [6, 16, 17, 20], "have": [6, 11, 12, 16, 17, 19], "guarante": 6, "column_name_with_id": [6, 16], "keep_old_id": [6, 16], "drop_na": 6, "ha": [6, 16, 17, 20], "must": [6, 16], "one": [6, 8, 16, 17], "doesn": 6, "t": [6, 16, 17, 19], "old": [6, 16], "drop": [6, 16, 17], "row": [6, 16, 17], "na": 6, "ad": [6, 16, 17], "input": [6, 8], "unless": 6, "zfish_gene_typ": [6, 17], "keep_missing_ortho": [6, 17], "retain": 6, "miss": 6, "target": 6, "zebrafish_id": [6, 17], "get_ortho_id": 6, "filenotfounderror": 6, "both": [6, 11, 16, 17], "raw_data": 6, "folder": [6, 16, 17], "work": [6, 12, 16, 17, 18], "subdirectori": [6, 16, 17], "zfin_to_ncbi_v": 6, "version_num": [6, 7], "txt": [6, 16, 17], "zfin_to_ensembl_v": 6, "merg": [6, 13], "save": [6, 16, 17], "master_gene_mapping_file_v": 6, "sure": 6, "befor": [6, 12, 16, 17], "zfish_human_orthology_v": 6, "zfish_human_orthologi": 6, "remov": [6, 11], "unnecessari": 6, "duplic": [6, 17], "entri": [6, 16, 17], "master_ortho_mapping_file_v": 6, "1": [7, 11, 16, 17, 18], "file_dir": 7, "database_dir": 7, "raw_data_dir": 7, "kegg_data_dir": 7, "go_data_dir": 7, "master_mapping_file_path": 7, "master_ortho_file_path": 7, "invalidorganismerror": 8, "check_valid_organ": 8, "chosen": 8, "string": 8, "normalize_organism_nam": 8, "organism_nam": 8, "normal": 8, "check_valid_zebrafish_gene_id_typ": 8, "singl": 8, "multipl": 8, "case": 8, "spell": [8, 16, 17], "sensit": [8, 16, 17], "check_valid_human_gene_id_typ": 8, "normalize_gene_id_typ": 8, "pretty_print_seri": [8, 16, 17], "save_data": [8, 16, 17], "locat": [8, 16, 17, 20], "download_url": 8, "url_to_download": 8, "download_url_to_fil": 8, "output_file_path": 8, "danrerlib": [9, 10, 12, 16, 17, 18, 20], "first": [10, 16, 17], "releas": [10, 12, 19], "In": [11, 16, 17], "foster": 11, "open": [11, 12], "welcom": [11, 12, 19], "environ": [11, 18], "we": [11, 16, 17, 18, 19], "contributor": 11, "maintain": [11, 16, 17], "particip": 11, "commun": [11, 14, 19], "harass": 11, "free": [11, 12, 13, 16, 17], "experi": 11, "everyon": 11, "regardless": 11, "ag": 11, "bodi": 11, "size": 11, "disabl": 11, "ethnic": 11, "gender": 11, "ident": 11, "express": [11, 13], "level": 11, "nation": 11, "person": [11, 13], "appear": 11, "race": 11, "religion": 11, "sexual": 11, "orient": 11, "behavior": 11, "contribut": [11, 14, 18], "posit": 11, "languag": 11, "Being": 11, "respect": [11, 18], "viewpoint": 11, "gracefulli": 11, "accept": 11, "construct": 11, "critic": 11, "focus": [11, 14], "what": [11, 16, 17, 20], "best": [11, 14, 16, 17, 18], "show": 11, "empathi": 11, "toward": 11, "member": 11, "unaccept": 11, "imageri": 11, "unwelcom": 11, "attent": 11, "advanc": 11, "troll": 11, "insult": 11, "derogatori": 11, "comment": 11, "polit": 11, "attack": 11, "public": 11, "privat": 11, "publish": [11, 13], "physic": 11, "electron": 11, "address": 11, "without": [11, 13, 16, 17], "explicit": 11, "permiss": [11, 13], "could": [11, 16, 17], "reason": 11, "consid": 11, "inappropri": 11, "profession": 11, "clarifi": 11, "expect": [11, 17, 20], "appropri": [11, 12], "fair": 11, "correct": [11, 20], "action": [11, 13], "ani": [11, 12, 13, 16, 17, 19, 20], "instanc": 11, "right": [11, 13], "edit": 11, "reject": 11, "commit": [11, 12], "wiki": 11, "issu": [11, 12, 16, 17, 19], "align": [11, 19], "ban": 11, "temporarili": 11, "perman": 11, "thei": [11, 12, 16], "deem": 11, "threaten": 11, "offens": 11, "harm": 11, "appli": 11, "within": [11, 20], "space": 11, "individu": 11, "its": [11, 12], "offici": [11, 12, 19], "mail": 11, "post": [11, 12], "via": 11, "social": 11, "media": 11, "account": 11, "act": 11, "appoint": 11, "onlin": 11, "offlin": 11, "event": [11, 13], "represent": [11, 20], "further": [11, 17, 18, 20], "defin": [11, 16, 17], "abus": 11, "report": [11, 17], "contact": [11, 19], "team": 11, "review": [11, 19], "investig": [11, 17], "complaint": 11, "respond": 11, "wai": [11, 16], "circumst": 11, "oblig": 11, "confidenti": 11, "regard": 11, "incid": 11, "polici": 11, "separ": [11, 16, 17], "who": 11, "do": [11, 13, 14, 16, 17], "good": 11, "faith": 11, "face": 11, "temporari": 11, "repercuss": 11, "leadership": 11, "adapt": 11, "coven": 11, "homepag": 11, "4": [11, 16, 17], "greatli": 12, "appreci": 12, "everi": [12, 17], "littl": 12, "bit": 12, "help": 12, "credit": 12, "alwai": [12, 16, 17], "oper": 12, "system": [12, 19], "about": [12, 14, 16, 18], "local": 12, "setup": 12, "might": [12, 16, 17], "troubleshoot": 12, "reproduc": 12, "github": 12, "anyth": 12, "tag": 12, "want": [12, 14, 16, 17, 19], "whoever": 12, "never": 12, "enough": 12, "feel": [12, 16, 17], "part": 12, "doc": 12, "docstr": 12, "even": [12, 16], "web": 12, "blog": 12, "articl": 12, "propos": 12, "explain": 12, "how": [12, 14, 16, 17, 19], "scope": 12, "narrow": 12, "possibl": [12, 20], "easier": 12, "rememb": [12, 16], "volunt": 12, "driven": 12, "readi": 12, "here": [12, 16, 17], "up": [12, 14, 17, 19], "develop": [12, 20], "copi": [12, 13, 16, 17], "instal": [12, 14, 16, 17, 19], "poetri": 12, "git": 12, "similar": 12, "branch": 12, "chang": [12, 14, 16, 17, 18], "checkout": 12, "b": [12, 16, 17], "bugfix": 12, "conform": 12, "pass": 12, "meet": 12, "addit": 12, "function": [12, 14, 16, 17, 19, 20], "python": [12, 14, 16, 17, 18, 19, 20], "By": [12, 16, 17], "agre": 12, "abid": 12, "copyright": 13, "c": 13, "2023": [13, 16, 17], "ashlei": [13, 16, 17], "schwartz": [13, 16, 17], "herebi": 13, "grant": 13, "charg": 13, "softwar": [13, 18], "deal": [13, 16, 17], "restrict": 13, "limit": [13, 16], "modifi": 13, "distribut": [13, 19], "sublicens": 13, "sell": 13, "permit": 13, "whom": 13, "furnish": 13, "subject": 13, "condit": 13, "abov": [13, 16, 17, 20], "notic": [13, 17], "shall": 13, "substanti": 13, "portion": 13, "THE": 13, "AS": 13, "warranti": 13, "OF": 13, "kind": 13, "OR": 13, "impli": 13, "BUT": 13, "NOT": 13, "TO": 13, "merchant": 13, "fit": 13, "FOR": 13, "particular": 13, "purpos": [13, 15, 18], "AND": 13, "noninfring": 13, "IN": 13, "NO": 13, "author": [13, 16, 17], "holder": 13, "BE": 13, "liabl": 13, "claim": 13, "damag": 13, "liabil": 13, "contract": 13, "tort": 13, "aris": 13, "out": [13, 14, 16, 17], "connect": 13, "WITH": 13, "short": [14, 20], "librari": [14, 16, 17, 18], "toolkit": [14, 20], "transcriptom": [14, 19], "our": [14, 16, 17, 19], "learn": [14, 19], "featur": [14, 20], "explor": [14, 19], "discov": 14, "most": [14, 16, 19], "instruct": [14, 16, 17], "principl": 14, "core": 14, "practic": [14, 18], "effect": 14, "modul": [14, 16, 17, 20], "home": 14, "hand": 14, "changelog": 14, "stai": 14, "date": [14, 16, 17, 19], "latest": [14, 19], "join": 14, "familiar": 14, "yourself": 14, "licens": 14, "execut": 15, "conclus": 15, "septemb": [16, 17], "goe": [16, 17], "over": [16, 17, 20], "simpli": 16, "anoth": [16, 17], "come": 16, "veri": 16, "form": 16, "depend": [16, 18, 20], "get": 16, "confus": 16, "start": [16, 17, 20], "zdb": [16, 17], "zebafish": [16, 17], "011219": [16, 17], "link": [16, 17], "integ": [16, 17, 20], "manag": [16, 17, 18, 20], "known": [16, 17, 20], "entrez": [16, 17, 20], "140634": [16, 17], "refseq": [16, 17], "cyp1a": [16, 17], "nomenclatur": [16, 17, 20], "ensdar": [16, 17, 20], "ensdarg00000098315": [16, 17], "util": [16, 17, 18, 20], "kei": [16, 17], "element": [16, 17], "sampl": [16, 17], "csv": [16, 17], "tsv": [16, 17], "past": [16, 17], "test_data": [16, 17], "rel": [16, 17], "path": [16, 17], "example_diff_express_data": [16, 17], "see": [16, 17, 18, 20], "gener": [16, 17, 18], "while": [16, 17], "need": [16, 17, 18], "larg": [16, 17], "foundat": [16, 17], "understand": [16, 17], "absolut": [16, 17], "There": [16, 17], "varieti": [16, 17, 20], "scenario": [16, 17], "simplest": 16, "few": [16, 17, 19], "sinc": 16, "common": 16, "entir": [16, 17], "excel": [16, 17], "fil": 16, "small": [16, 17], "especi": [16, 17], "just": [16, 17, 19], "list_of_gene_id": 16, "100000252": 16, "100000750": 16, "100001198": 16, "100001260": 16, "100002225": 16, "100002263": [16, 17], "100002756": 16, "100003223": 16, "100007521": 16, "100149273": 16, "100149794": 16, "100170795": 16, "100321746": 16, "100329897": 16, "100330617": 16, "2": [16, 17, 18], "tell": [16, 17], "program": [16, 17], "begin": [16, 20], "don": [16, 17, 19], "worri": 16, "let": [16, 17], "know": [16, 17], "made": 16, "mistak": 16, "launch": [16, 17], "current_gene_id_typ": 16, "desired_gene_id_typ": 16, "mean": [16, 17], "onc": 16, "converted_id": 16, "variabl": [16, 17], "visual": [16, 17], "print": [16, 17], "shell": [16, 17], "fine": 16, "idea": 16, "print_series_pretti": [16, 17], "030131": [16, 17], "1904": 16, "3404": 16, "030325": 16, "030616": 16, "609": 16, "040426": [16, 17], "743": 16, "050309": 16, "246": 16, "071009": 16, "080219": 16, "34": 16, "080723": 16, "44": 16, "081223": 16, "090313": 16, "141": 16, "091117": 16, "28": 16, "110309": 16, "120215": 16, "92": 16, "rather": [16, 17, 18], "previous": [16, 17], "choic": [16, 17], "file_nam": [16, 17], "out_data": [16, 17], "easili": [16, 17, 19], "repeat": [16, 17], "data_file_path": [16, 17], "ncbi_gene_list": 16, "read_csv": [16, 17], "sep": [16, 17], "30301": 16, "30459": 16, "57924": 16, "564772": 16, "794572": 16, "000616": 16, "10": [16, 17], "050419": 16, "85": 16, "080415": 16, "991229": 16, "12": 16, "manner": 16, "As": [16, 17], "now": 16, "below": [16, 17, 18], "non": 16, "quit": 16, "than": [16, 17, 18], "sometim": 16, "along": 16, "next": 16, "tabl": [16, 17], "am": [16, 17], "my": [16, 17, 18], "converst": 16, "activ": [16, 18, 20], "earlier": 16, "converted_id_t": 16, "7": [16, 17], "8": [16, 17], "9": [16, 17, 18], "11": 16, "13": [16, 18], "head": [16, 17], "automat": 16, "besid": 16, "initi": 16, "methodologi": 16, "great": 16, "where": [16, 17, 20], "properli": [16, 20], "sort": 16, "extrem": 16, "x": [16, 17], "y": [16, 17], "z": [16, 17], "pvalu": [16, 17], "logfc": [16, 17], "03": [16, 17], "essenti": [16, 17, 19], "geneid": [16, 17], "often": [16, 17], "lot": [16, 17], "sub": [16, 17], "extens": [16, 17], "realli": [16, 17], "quick": [16, 17, 19], "three": [16, 17], "stat": [16, 17], "line": [16, 17], "col": [16, 17], "shape": [16, 17], "f": [16, 17], "5464": [16, 17], "give": [16, 17], "new_data": [16, 17], "100000006": [16, 17], "5654": 16, "792615": [16, 17], "115009": [16, 17], "100000044": [16, 17], "121214": 16, "60": 16, "015286": [16, 17], "803879": [16, 17], "100000085": [16, 17], "1312": 16, "264762": [16, 17], "267360": [16, 17], "converted_datafram": 16, "remind": [16, 17], "test_set_invalid_col_nam": 16, "100000009": 16, "607285": 16, "144714": 16, "100000026": 16, "021338": 16, "603871": 16, "around": 16, "exactli": 16, "Of": 16, "cours": 16, "convent": 16, "seen": 16, "130530": 16, "778": 16, "120823": 16, "wa": [16, 18], "successfulli": 16, "though": [16, 17], "final": 16, "anymor": 16, "therefor": [16, 17, 18], "longer": 16, "conclud": [16, 17], "summari": [16, 17], "vice": 17, "versa": 17, "four": 17, "avail": 17, "1543": 17, "nci": 17, "hsa04911": 17, "dre04910": 17, "alreadi": [17, 18], "gather": 17, "mechan": 17, "been": [17, 20], "yet": 17, "approxim": 17, "70": 17, "advantag": 17, "those": 17, "list_of_zfish_id": 17, "chose": 17, "suffic": 17, "081113": 17, "031002": 17, "50": 17, "120104": 17, "000607": 17, "16": 17, "060503": 17, "934": 17, "050320": 17, "72": 17, "3147": 17, "2551": 17, "aka": 17, "4293": 17, "81553": 17, "26207": 17, "605": 17, "4861": 17, "55022": 17, "221143": 17, "save_seri": 17, "again": 17, "list_of_human_id": 17, "55585": 17, "23191": 17, "4192": 17, "5686": 17, "197021": 17, "390": 17, "344805": 17, "2623": 17, "zfish_desired_gene_id_typ": 17, "mdkb": 17, "cyfip1": 17, "psma5": 17, "lctla": 17, "rnd3b": 17, "ube2q1": 17, "gata1a": 17, "mdka": 17, "lctlb": 17, "gata1b": 17, "tmprss7": 17, "closer": 17, "end": 17, "state": 17, "fact": 17, "better": 17, "well": 17, "shown": 17, "file_path": 17, "human_ids_with_map": 17, "nan": 17, "becaus": 17, "downstream": 17, "isn": 17, "dropna": 17, "Or": 17, "ran": 17, "nearli": 17, "kept": 17, "04910": 17, "180": 17, "m": [17, 18], "len": 17, "117": 17, "left": 17, "had": 17, "recal": 17, "dre05910_human_gen": 17, "04911": 17, "n": [17, 18], "86": 17, "desired_zfish_gene_typ": 17, "zfish_id": 17, "64272": 17, "64269": 17, "64267": 17, "dtype": 17, "object": 17, "112": 17, "alpha": 17, "beta": 17, "someth": 17, "140615": 17, "9415": 17, "add_mapped_ortholg_column": 17, "84561": 17, "968": 17, "ortholog_datafram": 17, "went": 17, "113": 17, "built": 18, "proper": 18, "26": 18, "own": 18, "guidelin": [18, 19], "virtual": 18, "complic": 18, "anaconda": 18, "elsewher": 18, "decis": 18, "On": 18, "window": 18, "maco": 18, "linux": 18, "mention": 18, "app": 18, "task": [18, 19], "env": 18, "venv": 18, "mkdir": 18, "src": 18, "cd": 18, "bin": 18, "sourc": 18, "assist": 19, "annot": [19, 20], "resourc": 19, "guid": 19, "minut": 19, "eas": 19, "simplifi": 19, "clear": 19, "regularli": 19, "improv": 19, "encount": 19, "question": 19, "hesit": 19, "u": 19, "under": [19, 20], "mit": 19, "drl": 20, "typic": 20, "wish": 20, "introduct": 20, "referenc": 20, "IF": 20, "websit": 20, "network": 20, "alphabet": 20, "mor": 20, "uniqu": 20, "curat": 20, "biochem": 20, "catalog": 20, "standard": 20, "vocabulari": 20, "definit": 20, "subcellular": 20, "structur": 20, "underrepres": 20, "chanc": 20, "info": 20}, "objects": {"": [[5, 0, 0, "-", "danrerlib"]], "danrerlib": [[1, 0, 0, "-", "GO"], [2, 0, 0, "-", "KEGG"], [3, 0, 0, "-", "database"], [4, 0, 0, "-", "enrichment"], [6, 0, 0, "-", "mapping"], [7, 0, 0, "-", "settings"], [8, 0, 0, "-", "utils"]], "danrerlib.GO": [[1, 1, 1, "", "build_gene_ontology_database"], [1, 1, 1, "", "get_genes_in_GO_concept"], [1, 1, 1, "", "id_exists_given_organism"]], "danrerlib.KEGG": [[2, 1, 1, "", "build_kegg_database"], [2, 1, 1, "", "get_genes_in_disease"], [2, 1, 1, "", "get_genes_in_pathway"]], "danrerlib.enrichment": [[4, 1, 1, "", "enrich_GO"], [4, 1, 1, "", "enrich_KEGG"], [4, 1, 1, "", "fishers"], [4, 1, 1, "", "logistic"]], "danrerlib.mapping": [[6, 2, 1, "", "DatabaseNotFoundError"], [6, 2, 1, "", "InvalidGeneTypeError"], [6, 1, 1, "", "add_mapped_column"], [6, 1, 1, "", "add_mapped_ortholog_column"], [6, 1, 1, "", "build_gene_mapping"], [6, 1, 1, "", "build_ortho_mapping"], [6, 1, 1, "", "convert_ids"], [6, 1, 1, "", "convert_to_human"], [6, 1, 1, "", "convert_to_zebrafish"], [6, 1, 1, "", "get_ortho_ids"]], "danrerlib.settings": [[7, 3, 1, "", "DATABASE_DIR"], [7, 3, 1, "", "ENS_ID"], [7, 3, 1, "", "FILE_DIR"], [7, 3, 1, "", "GO_DATA_DIR"], [7, 3, 1, "", "HUMAN_ID"], [7, 3, 1, "", "KEGG_DATA_DIR"], [7, 3, 1, "", "MASTER_MAPPING_FILE_PATH"], [7, 3, 1, "", "MASTER_ORTHO_FILE_PATH"], [7, 3, 1, "", "NCBI_ID"], [7, 3, 1, "", "RAW_DATA_DIR"], [7, 3, 1, "", "SYMBOL"], [7, 3, 1, "", "VERSION_NUM"], [7, 3, 1, "", "ZFIN_ID"]], "danrerlib.utils": [[8, 2, 1, "", "InvalidGeneTypeError"], [8, 2, 1, "", "InvalidOrganismError"], [8, 1, 1, "", "check_valid_human_gene_id_type"], [8, 1, 1, "", "check_valid_organism"], [8, 1, 1, "", "check_valid_zebrafish_gene_id_type"], [8, 1, 1, "", "download_url"], [8, 1, 1, "", "download_url_to_file"], [8, 1, 1, "", "normalize_gene_id_type"], [8, 1, 1, "", "normalize_organism_name"], [8, 1, 1, "", "pretty_print_series"], [8, 1, 1, "", "save_data"]]}, "objtypes": {"0": "py:module", "1": "py:function", "2": "py:exception", "3": "py:data"}, "objnames": {"0": ["py", "module", "Python module"], "1": ["py", "function", "Python function"], "2": ["py", "exception", "Python exception"], "3": ["py", "data", "Python data"]}, "titleterms": {"api": [0, 9, 14], "refer": [0, 9, 14], "danrerlib": [1, 2, 3, 4, 5, 6, 7, 8, 14, 19], "go": 1, "modul": [1, 2, 4, 6, 7, 8], "content": [1, 2, 4, 6, 7, 8], "function": [1, 2, 4, 6, 8], "kegg": 2, "databas": [3, 20], "enrich": [4, 20], "submodul": 5, "map": [6, 16, 17], "set": 7, "util": 8, "changelog": 10, "v0": 10, "1": 10, "0": 10, "21": 10, "09": 10, "2023": 10, "code": [11, 12], "conduct": [11, 12], "our": 11, "pledg": 11, "standard": 11, "respons": 11, "scope": 11, "enforc": 11, "attribut": 11, "contribut": [12, 19], "type": [12, 20], "report": 12, "bug": 12, "fix": 12, "implement": 12, "featur": [12, 19], "write": 12, "document": [12, 14], "submit": 12, "feedback": [12, 19], "get": [12, 14, 17, 19], "start": [12, 14, 19], "pull": 12, "request": 12, "guidelin": 12, "licens": [13, 19], "mit": 13, "welcom": 14, "": 14, "introduct": [14, 19], "user": [14, 18], "guid": [14, 18], "tutori": [14, 15], "develop": [14, 18], "gene": [16, 17, 20], "id": [16, 20], "purpos": [16, 17], "background": [16, 17], "requir": [16, 17], "execut": [16, 17], "simpl": [16, 17], "case": [16, 17], "convert": 16, "list": 16, "from": [16, 17], "file": [16, 17], "keep": [16, 17], "column": [16, 17], "larger": [16, 17], "dataset": [16, 17], "special": 16, "conclus": [16, 17], "orthologi": 17, "ortholog": 17, "zebrafish": [17, 20], "human": [17, 20], "instal": 18, "quick": 18, "conda": 18, "pypi": 18, "pip": 18, "version": 18, "begin": 18, "advanc": 18, "recommend": 18, "method": 18, "kei": 19, "why": 19, "support": 19, "usag": 20, "principl": 20, "workflow": 20, "pathwai": 20}, "envversion": {"sphinx.domains.c": 2, "sphinx.domains.changeset": 1, "sphinx.domains.citation": 1, "sphinx.domains.cpp": 8, "sphinx.domains.index": 1, "sphinx.domains.javascript": 2, "sphinx.domains.math": 2, "sphinx.domains.python": 3, "sphinx.domains.rst": 2, "sphinx.domains.std": 2, "sphinx": 57}, "alltitles": {"API Reference": [[0, "api-reference"], [9, "api-reference"], [14, "api-reference"]], "danrerlib.GO": [[1, "module-danrerlib.GO"]], "Module Contents": [[1, "module-contents"], [2, "module-contents"], [4, "module-contents"], [6, "module-contents"], [7, "module-contents"], [8, "module-contents"]], "Functions": [[1, "functions"], [2, "functions"], [4, "functions"], [6, "functions"], [8, "functions"]], "danrerlib.KEGG": [[2, "module-danrerlib.KEGG"]], "danrerlib.database": [[3, "module-danrerlib.database"]], "danrerlib.enrichment": [[4, "module-danrerlib.enrichment"]], "danrerlib": [[5, "module-danrerlib"]], "Submodules": [[5, "submodules"]], "danrerlib.mapping": [[6, "module-danrerlib.mapping"]], "danrerlib.settings": [[7, "module-danrerlib.settings"]], "danrerlib.utils": [[8, "module-danrerlib.utils"]], "Changelog": [[10, "changelog"]], "v0.1.0 (21/09/2023)": [[10, "v0-1-0-21-09-2023"]], "Code of Conduct": [[11, "code-of-conduct"], [12, "code-of-conduct"]], "Our Pledge": [[11, "our-pledge"]], "Our Standards": [[11, "our-standards"]], "Our Responsibilities": [[11, "our-responsibilities"]], "Scope": [[11, "scope"]], "Enforcement": [[11, "enforcement"]], "Attribution": [[11, "attribution"]], "Contributing": [[12, "contributing"], [19, "contributing"]], "Types of Contributions": [[12, "types-of-contributions"]], "Report Bugs": [[12, "report-bugs"]], "Fix Bugs": [[12, "fix-bugs"]], "Implement Features": [[12, "implement-features"]], "Write Documentation": [[12, "write-documentation"]], "Submit Feedback": [[12, "submit-feedback"]], "Get Started!": [[12, "get-started"]], "Pull Request Guidelines": [[12, "pull-request-guidelines"]], "License": [[13, "license"], [19, "license"]], "MIT License": [[13, "mit-license"]], "Welcome to danRerLib\u2019s Documentation": [[14, "welcome-to-danrerlib-s-documentation"]], "Introduction": [[14, "introduction"], [19, "introduction"]], "Getting Started": [[14, "getting-started"], [19, "getting-started"]], "User\u2019s Guide": [[14, "user-s-guide"]], "Tutorials": [[14, "tutorials"], [15, "tutorials"]], "Developer Guide": [[14, "developer-guide"]], "Gene ID Mapping": [[16, "gene-id-mapping"]], "Purpose and Background": [[16, "purpose-and-background"], [17, "purpose-and-background"]], "Requirements": [[16, "requirements"], [17, "requirements"]], "Execute Mappings": [[16, "execute-mappings"]], "Simple Case: Convert a list of Gene IDs": [[16, "simple-case-convert-a-list-of-gene-ids"]], "Simple Case: Convert a list of Gene IDs From a File": [[16, "simple-case-convert-a-list-of-gene-ids-from-a-file"]], "Simple Case: Convert a list of Gene IDs and Keep Mapping": [[16, "simple-case-convert-a-list-of-gene-ids-and-keep-mapping"]], "Convert Gene IDs in a Column of a Larger Dataset": [[16, "convert-gene-ids-in-a-column-of-a-larger-dataset"]], "Special Cases": [[16, "special-cases"]], "Conclusion": [[16, "conclusion"], [17, "conclusion"]], "Orthology Mapping": [[17, "orthology-mapping"]], "Execute Orthology Mappings": [[17, "execute-orthology-mappings"]], "Simple Case: Get orthologs": [[17, "simple-case-get-orthologs"]], "Zebrafish to Human": [[17, "zebrafish-to-human"], [17, "id1"], [17, "id3"]], "Human to Zebrafish": [[17, "human-to-zebrafish"], [17, "id2"], [17, "id4"]], "Simple Case: Get Orthologs and Keep Mapping": [[17, "simple-case-get-orthologs-and-keep-mapping"]], "Get Orthologs for Genes from File": [[17, "get-orthologs-for-genes-from-file"]], "Get Orthologs for a Column of a Larger Dataset": [[17, "get-orthologs-for-a-column-of-a-larger-dataset"]], "Installation": [[18, "installation"]], "Quick Install": [[18, "quick-install"]], "Conda": [[18, "conda"]], "PyPi/pip": [[18, "pypi-pip"]], "Development Version": [[18, "development-version"]], "Installation Guide": [[18, "installation-guide"]], "Beginning users": [[18, "beginning-users"]], "Advanced users (recommended method)": [[18, "advanced-users-recommended-method"]], "Key Features": [[19, "key-features"]], "Why danRerLib?": [[19, "why-danrerlib"]], "Support and Feedback": [[19, "support-and-feedback"]], "Usage Principles": [[20, "usage-principles"]], "Workflow": [[20, "workflow"]], "Zebrafish Gene ID Types": [[20, "zebrafish-gene-id-types"]], "Human Gene ID Type": [[20, "human-gene-id-type"]], "Pathway Databases": [[20, "pathway-databases"]], "Enrichment Types": [[20, "enrichment-types"]]}, "indexentries": {"build_gene_ontology_database() (in module danrerlib.go)": [[1, "danrerlib.GO.build_gene_ontology_database"]], "danrerlib.go": [[1, "module-danrerlib.GO"]], "get_genes_in_go_concept() (in module danrerlib.go)": [[1, "danrerlib.GO.get_genes_in_GO_concept"]], "id_exists_given_organism() (in module danrerlib.go)": [[1, "danrerlib.GO.id_exists_given_organism"]], "module": [[1, "module-danrerlib.GO"], [2, "module-danrerlib.KEGG"], [3, "module-danrerlib.database"], [4, "module-danrerlib.enrichment"], [5, "module-danrerlib"], [6, "module-danrerlib.mapping"], [7, "module-danrerlib.settings"], [8, "module-danrerlib.utils"]], "build_kegg_database() (in module danrerlib.kegg)": [[2, "danrerlib.KEGG.build_kegg_database"]], "danrerlib.kegg": [[2, "module-danrerlib.KEGG"]], "get_genes_in_disease() (in module danrerlib.kegg)": [[2, "danrerlib.KEGG.get_genes_in_disease"]], "get_genes_in_pathway() (in module danrerlib.kegg)": [[2, "danrerlib.KEGG.get_genes_in_pathway"]], "danrerlib.database": [[3, "module-danrerlib.database"]], "danrerlib.enrichment": [[4, "module-danrerlib.enrichment"]], "enrich_go() (in module danrerlib.enrichment)": [[4, "danrerlib.enrichment.enrich_GO"]], "enrich_kegg() (in module danrerlib.enrichment)": [[4, "danrerlib.enrichment.enrich_KEGG"]], "fishers() (in module danrerlib.enrichment)": [[4, "danrerlib.enrichment.fishers"]], "logistic() (in module danrerlib.enrichment)": [[4, "danrerlib.enrichment.logistic"]], "danrerlib": [[5, "module-danrerlib"]], "databasenotfounderror": [[6, "danrerlib.mapping.DatabaseNotFoundError"]], "invalidgenetypeerror": [[6, "danrerlib.mapping.InvalidGeneTypeError"], [8, "danrerlib.utils.InvalidGeneTypeError"]], "add_mapped_column() (in module danrerlib.mapping)": [[6, "danrerlib.mapping.add_mapped_column"]], "add_mapped_ortholog_column() (in module danrerlib.mapping)": [[6, "danrerlib.mapping.add_mapped_ortholog_column"]], "build_gene_mapping() (in module danrerlib.mapping)": [[6, "danrerlib.mapping.build_gene_mapping"]], "build_ortho_mapping() (in module danrerlib.mapping)": [[6, "danrerlib.mapping.build_ortho_mapping"]], "convert_ids() (in module danrerlib.mapping)": [[6, "danrerlib.mapping.convert_ids"]], "convert_to_human() (in module danrerlib.mapping)": [[6, "danrerlib.mapping.convert_to_human"]], "convert_to_zebrafish() (in module danrerlib.mapping)": [[6, "danrerlib.mapping.convert_to_zebrafish"]], "danrerlib.mapping": [[6, "module-danrerlib.mapping"]], "get_ortho_ids() (in module danrerlib.mapping)": [[6, "danrerlib.mapping.get_ortho_ids"]], "database_dir (in module danrerlib.settings)": [[7, "danrerlib.settings.DATABASE_DIR"]], "ens_id (in module danrerlib.settings)": [[7, "danrerlib.settings.ENS_ID"]], "file_dir (in module danrerlib.settings)": [[7, "danrerlib.settings.FILE_DIR"]], "go_data_dir (in module danrerlib.settings)": [[7, "danrerlib.settings.GO_DATA_DIR"]], "human_id (in module danrerlib.settings)": [[7, "danrerlib.settings.HUMAN_ID"]], "kegg_data_dir (in module danrerlib.settings)": [[7, "danrerlib.settings.KEGG_DATA_DIR"]], "master_mapping_file_path (in module danrerlib.settings)": [[7, "danrerlib.settings.MASTER_MAPPING_FILE_PATH"]], "master_ortho_file_path (in module danrerlib.settings)": [[7, "danrerlib.settings.MASTER_ORTHO_FILE_PATH"]], "ncbi_id (in module danrerlib.settings)": [[7, "danrerlib.settings.NCBI_ID"]], "raw_data_dir (in module danrerlib.settings)": [[7, "danrerlib.settings.RAW_DATA_DIR"]], "symbol (in module danrerlib.settings)": [[7, "danrerlib.settings.SYMBOL"]], "version_num (in module danrerlib.settings)": [[7, "danrerlib.settings.VERSION_NUM"]], "zfin_id (in module danrerlib.settings)": [[7, "danrerlib.settings.ZFIN_ID"]], "danrerlib.settings": [[7, "module-danrerlib.settings"]], "invalidorganismerror": [[8, "danrerlib.utils.InvalidOrganismError"]], "check_valid_human_gene_id_type() (in module danrerlib.utils)": [[8, "danrerlib.utils.check_valid_human_gene_id_type"]], "check_valid_organism() (in module danrerlib.utils)": [[8, "danrerlib.utils.check_valid_organism"]], "check_valid_zebrafish_gene_id_type() (in module danrerlib.utils)": [[8, "danrerlib.utils.check_valid_zebrafish_gene_id_type"]], "danrerlib.utils": [[8, "module-danrerlib.utils"]], "download_url() (in module danrerlib.utils)": [[8, "danrerlib.utils.download_url"]], "download_url_to_file() (in module danrerlib.utils)": [[8, "danrerlib.utils.download_url_to_file"]], "normalize_gene_id_type() (in module danrerlib.utils)": [[8, "danrerlib.utils.normalize_gene_id_type"]], "normalize_organism_name() (in module danrerlib.utils)": [[8, "danrerlib.utils.normalize_organism_name"]], "pretty_print_series() (in module danrerlib.utils)": [[8, "danrerlib.utils.pretty_print_series"]], "save_data() (in module danrerlib.utils)": [[8, "danrerlib.utils.save_data"]]}})