<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Gene ID Mapping &mdash; danrerlib</title>
      <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="../_static/mystnb.4510f1fc1dee50b3e5859aac5469c37c29e427902b24a333a5f9fcb2f0b3ac41.css" type="text/css" />
      <link rel="stylesheet" href="../_static/graphviz.css" type="text/css" />
      <link rel="stylesheet" href="../_static/custom.css" type="text/css" />
      <link rel="stylesheet" href="../_static/fonts.css" type="text/css" />
      <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script data-url_root="../" id="documentation_options" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/sphinx_highlight.js"></script>
    <script src="../_static/js/theme.js"></script>
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Orthology Mapping" href="perform_orthology_mapping.html" />
    <link rel="prev" title="danrerlib.mapping" href="../autoapi/danrerlib/mapping/index.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="../index.html" class="icon icon-home">
            danrerlib
              <img src="../_static/danrerlib_logo.png" class="logo" alt="Logo"/>
          </a>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">User's Guide</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../userguide/introduction.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../userguide/installation.html">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../userguide/usage.html">Usage Principles</a></li>
<li class="toctree-l1"><a class="reference internal" href="../autoapi/index.html">API Reference</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Tutorials</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Gene ID Mapping</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#purpose-and-background">Purpose and Background</a></li>
<li class="toctree-l2"><a class="reference internal" href="#requirements">Requirements</a></li>
<li class="toctree-l2"><a class="reference internal" href="#execute-mappings">Execute Mappings</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#simple-case-convert-a-list-of-gene-ids">Simple Case: Convert a list of Gene IDs</a></li>
<li class="toctree-l3"><a class="reference internal" href="#simple-case-convert-a-list-of-gene-ids-from-a-file">Simple Case: Convert a list of Gene IDs From a File</a></li>
<li class="toctree-l3"><a class="reference internal" href="#simple-case-convert-a-list-of-gene-ids-and-keep-mapping">Simple Case: Convert a list of Gene IDs and Keep Mapping</a></li>
<li class="toctree-l3"><a class="reference internal" href="#convert-gene-ids-in-a-column-of-a-larger-dataset">Convert Gene IDs in a Column of a Larger Dataset</a><ul>
<li class="toctree-l4"><a class="reference internal" href="#special-cases">Special Cases</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#conclusion">Conclusion</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="perform_orthology_mapping.html">Orthology Mapping</a></li>
</ul>
<p class="caption" role="heading"><span class="caption-text">Developer's Guide</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../devguide/changelog.html">Changelog</a></li>
<li class="toctree-l1"><a class="reference internal" href="../devguide/contributing.html">Contributing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../devguide/conduct.html">Code of Conduct</a></li>
<li class="toctree-l1"><a class="reference internal" href="../devguide/license.html">License</a></li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">danrerlib</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="../index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Gene ID Mapping</li>
      <li class="wy-breadcrumbs-aside">
            <a href="../_sources/tutorials/perform_geneid_mapping.ipynb.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="gene-id-mapping">
<h1>Gene ID Mapping<a class="headerlink" href="#gene-id-mapping" title="Permalink to this heading"></a></h1>
<p>Author: Ashley Schwartz</p>
<p>Date: September 2023</p>
<section id="purpose-and-background">
<h2>Purpose and Background<a class="headerlink" href="#purpose-and-background" title="Permalink to this heading"></a></h2>
<p>This tutorial goes over how to simply convert a list of zebrafish Gene IDs to another Gene ID type. Gene IDs come in very different forms depending on the database or genome build you are using. This can get confusing! The Gene ID options are:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Gene ID Name</p></th>
<th class="head"><p>Description</p></th>
<th class="head"><p>Example</p></th>
<th class="head"><p>Notes</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>ZFIN ID</p></td>
<td><p>ZFIN gene id: always starts with ‘ZDB’ for zebafish database</p></td>
<td><p>ZDB-GENE-011219-1</p></td>
<td><p>used as the “master” gene id (<a class="reference external" href="https://zfin.org/">link</a>)</p></td>
</tr>
<tr class="row-odd"><td><p>NCBI Gene ID</p></td>
<td><p>integer gene id managed by NCBI: also known as Entrez Gene ID</p></td>
<td><p>140634</p></td>
<td><p><a class="reference external" href="https://www.ncbi.nlm.nih.gov/gene?Db=gene&amp;Cmd=DetailsSearch&amp;Term=140634">link</a></p></td>
</tr>
<tr class="row-even"><td><p>Symbol</p></td>
<td><p>descriptive symbol/name: RefSeq symbol used in RefSeq genome build</p></td>
<td><p>cyp1a</p></td>
<td><p>nomenclature defined by ZFIN</p></td>
</tr>
<tr class="row-odd"><td><p>Ensembl Gene ID</p></td>
<td><p>Ensembl database gene id: always starts with ‘ENSDAR’</p></td>
<td><p>ENSDARG00000098315</p></td>
<td><p><a class="reference external" href="http://useast.ensembl.org/Danio_rerio/Location/View?g=ENSDARG00000098315;r=18:5588068-5598958">link</a></p></td>
</tr>
</tbody>
</table>
</section>
<section id="requirements">
<h2>Requirements<a class="headerlink" href="#requirements" title="Permalink to this heading"></a></h2>
<p>In this tutorial we will be utilizing two key elements:</p>
<ul class="simple">
<li><p>a sample Gene ID list (format: .csv, .tsv, .txt) for reading in the Gene IDs, otherwise typing or copy/pasting Gene IDs is also supported</p>
<ul>
<li><p>the gene list we will be using is located in the data/test_data subdirectory of this current working directory with relative path <code class="docutils literal notranslate"><span class="pre">data/test_data/example_diff_express_data.txt</span></code></p></li>
</ul>
</li>
<li><p>the required python package</p>
<ul>
<li><p>see install notes if not currently installed.</p></li>
</ul>
</li>
</ul>
<p>In general, while you do not need a large foundation in Python to execute the code listed in this tutorial, a general understanding of absolute and relative paths is useful.</p>
<p><em>note: the Gene IDs are spelling and case sensitive</em></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># IMPORT PYTHON PACKAGE</span>
<span class="c1"># ---------------------</span>
<span class="kn">from</span> <span class="nn">danrerlib</span> <span class="kn">import</span> <span class="n">mapping</span><span class="p">,</span> <span class="n">utils</span>
<span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="execute-mappings">
<h2>Execute Mappings<a class="headerlink" href="#execute-mappings" title="Permalink to this heading"></a></h2>
<p>There are a variety of scenarios when you might need to map Gene IDs. In the most simplest case, you might have a few IDs you would like to map to Entrez NCBI Gene IDs since that is a common Gene ID used in pathway databases. Other times, you might want to convert an entire column in an excel fil you have. We will go through a few different options.</p>
<section id="simple-case-convert-a-list-of-gene-ids">
<h3>Simple Case: Convert a list of Gene IDs<a class="headerlink" href="#simple-case-convert-a-list-of-gene-ids" title="Permalink to this heading"></a></h3>
<p><em>Purpose: Given a small list of Gene IDs that are of Gene ID type A, convert to Gene ID type B.</em></p>
<p>You would most likely use the simple case if you have a small list of gene ids that you need to convert. Especially useful if you just want to copy and paste and retrieve your converted ids!</p>
<p><strong>Step 1</strong>: Define your list of ids. In this case, I have NCBI Gene IDs. I named the python list <code class="docutils literal notranslate"><span class="pre">list_of_gene_ids</span></code> and include all the Gene IDs I want to convert.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">list_of_gene_ids</span> <span class="o">=</span> <span class="p">[</span> 
    <span class="mi">100000252</span><span class="p">,</span> <span class="mi">100000750</span><span class="p">,</span> <span class="mi">100001198</span><span class="p">,</span> <span class="mi">100001260</span><span class="p">,</span> <span class="mi">100002225</span><span class="p">,</span> <span class="mi">100002263</span><span class="p">,</span> 
    <span class="mi">100002756</span><span class="p">,</span> <span class="mi">100003223</span><span class="p">,</span> <span class="mi">100007521</span><span class="p">,</span> <span class="mi">100149273</span><span class="p">,</span> <span class="mi">100149794</span><span class="p">,</span> <span class="mi">100170795</span><span class="p">,</span>
    <span class="mi">100321746</span><span class="p">,</span> <span class="mi">100329897</span><span class="p">,</span> <span class="mi">100330617</span><span class="p">,</span>
<span class="p">]</span>
</pre></div>
</div>
</div>
</div>
<p><strong>Step 2</strong>: Tell the program which ID you currently have and which ID you would like to convert to. I currently have NCBI Gene IDs and I want to convert to ZFIN Gene IDs. Note that Gene ID options are spelling and case sensitive. Options are listed at the beginning of this document. (don’t worry, the program will let you know if you have made a mistake when you launch the program!)</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">current_gene_id_type</span> <span class="o">=</span> <span class="s1">&#39;NCBI Gene ID&#39;</span>
<span class="n">desired_gene_id_type</span> <span class="o">=</span> <span class="s1">&#39;ZFIN ID&#39;</span>
</pre></div>
</div>
</div>
</div>
<p><strong>Step 3</strong>: Launch the conversion function to get your converted ids. This means we want to run the <code class="docutils literal notranslate"><span class="pre">convert_ids</span></code> function in the <code class="docutils literal notranslate"><span class="pre">mapping</span></code> module of our library. Once executed, the converted ids will be stored in the <code class="docutils literal notranslate"><span class="pre">converted_ids</span></code> variable.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># do conversion</span>
<span class="n">converted_ids</span> <span class="o">=</span> <span class="n">mapping</span><span class="o">.</span><span class="n">convert_ids</span><span class="p">(</span><span class="n">list_of_gene_ids</span><span class="p">,</span> <span class="n">current_gene_id_type</span><span class="p">,</span> <span class="n">desired_gene_id_type</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p><strong>Step 4</strong>: To visualize your converted ids, you can either print them to the python shell or save them to a file. If you would like to print them, which is a fine idea if you only have a few, you can use the <code class="docutils literal notranslate"><span class="pre">print_series_pretty</span></code> function in the <code class="docutils literal notranslate"><span class="pre">utils</span></code> module of the library.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">utils</span><span class="o">.</span><span class="n">pretty_print_series</span><span class="p">(</span><span class="n">converted_ids</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>ZDB-GENE-030131-1904
ZDB-GENE-030131-3404
ZDB-GENE-030325-1
ZDB-GENE-030616-609
ZDB-GENE-040426-743
ZDB-GENE-050309-246
ZDB-GENE-071009-6
ZDB-GENE-080219-34
ZDB-GENE-080723-44
ZDB-GENE-081223-2
ZDB-GENE-090313-141
ZDB-GENE-091117-28
ZDB-GENE-110309-3
ZDB-GENE-120215-92
</pre></div>
</div>
</div>
</div>
<p>If you would rather save the data to a file, you can save <code class="docutils literal notranslate"><span class="pre">converted_ids</span></code> to a file name called <code class="docutils literal notranslate"><span class="pre">converted_ids.txt</span></code> in the output data directory we defined previously. For some default options, you can use the <code class="docutils literal notranslate"><span class="pre">save_data</span></code> function in the <code class="docutils literal notranslate"><span class="pre">utils</span></code> module. Feel free to change the output directory to any folder of your choice.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">file_name</span> <span class="o">=</span> <span class="s1">&#39;data/out_data/converted_ids.txt&#39;</span>
<span class="n">utils</span><span class="o">.</span><span class="n">save_data</span><span class="p">(</span><span class="n">converted_ids</span><span class="p">,</span> <span class="n">file_name</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
</section>
<section id="simple-case-convert-a-list-of-gene-ids-from-a-file">
<h3>Simple Case: Convert a list of Gene IDs From a File<a class="headerlink" href="#simple-case-convert-a-list-of-gene-ids-from-a-file" title="Permalink to this heading"></a></h3>
<p><em>Purpose: Given a list of Gene IDs from a file that are of Gene ID type A, convert to Gene ID type B.</em></p>
<p>If you have a file that contains a list of Gene IDs, you can easily repeat these steps by reading in that file first. Check it out:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">data_file_path</span> <span class="o">=</span> <span class="s1">&#39;data/test_data/ncbi_gene_list.txt&#39;</span>
<span class="n">list_of_gene_ids</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">data_file_path</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">list_of_gene_ids</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>NCBI Gene ID</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>30301</td>
    </tr>
    <tr>
      <th>1</th>
      <td>30459</td>
    </tr>
    <tr>
      <th>2</th>
      <td>57924</td>
    </tr>
    <tr>
      <th>3</th>
      <td>564772</td>
    </tr>
    <tr>
      <th>4</th>
      <td>794572</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>When we use the <code class="docutils literal notranslate"><span class="pre">pandas</span></code> package to read in the data, it organizes it into what is called a <code class="docutils literal notranslate"><span class="pre">Pandas</span> <span class="pre">DataFrame</span></code>. All other steps can be executed the same way.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">current_gene_id_type</span> <span class="o">=</span> <span class="s1">&#39;NCBI Gene ID&#39;</span>
<span class="n">desired_gene_id_type</span> <span class="o">=</span> <span class="s1">&#39;ZFIN ID&#39;</span>

<span class="c1"># do conversion</span>
<span class="n">converted_ids</span> <span class="o">=</span> <span class="n">mapping</span><span class="o">.</span><span class="n">convert_ids</span><span class="p">(</span><span class="n">list_of_gene_ids</span><span class="p">,</span> <span class="n">current_gene_id_type</span><span class="p">,</span> <span class="n">desired_gene_id_type</span><span class="p">)</span>

<span class="n">utils</span><span class="o">.</span><span class="n">pretty_print_series</span><span class="p">(</span><span class="n">converted_ids</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>ZDB-GENE-000616-10
ZDB-GENE-050419-85
ZDB-GENE-080415-1
ZDB-GENE-991229-12
</pre></div>
</div>
</div>
</div>
<p>You could also save the data to a file in the same manner. As you can see, we now have our converted ids. The ids will be in order based on the original Gene IDs given to the program. If you would like to keep the mapping, adding a column to the gene IDs you currently have might be useful (see below). Limitations of this method include a non 1:1 mapping between Gene ID options. It is quite common that there is more than one Ensembl Gene ID for another Gene ID option. This function will return all mappings, but, since it just returns the list, you do not know which gene in your original set has mapped to two different genes in the new set. This may not be an issue for some use cases, but sometimes it is important to know.</p>
<p>If you would like to keep the old Gene IDs along with the mapping, follow the next set of instructions!</p>
</section>
<section id="simple-case-convert-a-list-of-gene-ids-and-keep-mapping">
<h3>Simple Case: Convert a list of Gene IDs and Keep Mapping<a class="headerlink" href="#simple-case-convert-a-list-of-gene-ids-and-keep-mapping" title="Permalink to this heading"></a></h3>
<p><em>Purpose: Given a list of Gene IDs that are of Gene ID type A, convert to Gene ID type B and keep both Gene ID A and Gene ID B in a table.</em></p>
<p><strong>Step 1</strong>: Define your list of ids. This is the same list I used above, and remember they are NCBI Gene IDs. I am also defining my current Gene ID type and the Gene ID type I would like to convert to here.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">list_of_gene_ids</span> <span class="o">=</span> <span class="p">[</span> 
    <span class="mi">100000252</span><span class="p">,</span> <span class="mi">100000750</span><span class="p">,</span> <span class="mi">100001198</span><span class="p">,</span> <span class="mi">100001260</span><span class="p">,</span> <span class="mi">100002225</span><span class="p">,</span> <span class="mi">100002263</span><span class="p">,</span> 
    <span class="mi">100002756</span><span class="p">,</span> <span class="mi">100003223</span><span class="p">,</span> <span class="mi">100007521</span><span class="p">,</span> <span class="mi">100149273</span><span class="p">,</span> <span class="mi">100149794</span><span class="p">,</span> <span class="mi">100170795</span><span class="p">,</span>
    <span class="mi">100321746</span><span class="p">,</span> <span class="mi">100329897</span><span class="p">,</span> <span class="mi">100330617</span><span class="p">,</span>
<span class="p">]</span>

<span class="n">current_gene_id_type</span> <span class="o">=</span> <span class="s1">&#39;NCBI Gene ID&#39;</span>
<span class="n">desired_gene_id_type</span> <span class="o">=</span> <span class="s1">&#39;ZFIN ID&#39;</span>
</pre></div>
</div>
</div>
</div>
<p><strong>Step 2:</strong> Do converstion. If we would like to keep the mapping, we would use the <code class="docutils literal notranslate"><span class="pre">convert_ids</span></code> function in the <code class="docutils literal notranslate"><span class="pre">mapping</span></code> module of our library but activate the <code class="docutils literal notranslate"><span class="pre">keep_mapping</span></code> parameter. By default, as used earlier, <code class="docutils literal notranslate"><span class="pre">keep_mapping</span> <span class="pre">=</span> <span class="pre">False</span></code></p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># do conversion</span>
<span class="n">converted_id_table</span> <span class="o">=</span> <span class="n">mapping</span><span class="o">.</span><span class="n">convert_ids</span><span class="p">(</span><span class="n">list_of_gene_ids</span><span class="p">,</span> <span class="n">current_gene_id_type</span><span class="p">,</span> <span class="n">desired_gene_id_type</span><span class="p">,</span> <span class="n">keep_mapping</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">converted_id_table</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>NCBI Gene ID</th>
      <th>ZFIN ID</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>100002263</td>
      <td>ZDB-GENE-030131-1904</td>
    </tr>
    <tr>
      <th>1</th>
      <td>100330617</td>
      <td>ZDB-GENE-030131-3404</td>
    </tr>
    <tr>
      <th>2</th>
      <td>100001198</td>
      <td>ZDB-GENE-030325-1</td>
    </tr>
    <tr>
      <th>3</th>
      <td>100003223</td>
      <td>ZDB-GENE-030616-609</td>
    </tr>
    <tr>
      <th>4</th>
      <td>100000252</td>
      <td>ZDB-GENE-040426-743</td>
    </tr>
    <tr>
      <th>5</th>
      <td>100001260</td>
      <td>ZDB-GENE-050309-246</td>
    </tr>
    <tr>
      <th>6</th>
      <td>100321746</td>
      <td>ZDB-GENE-071009-6</td>
    </tr>
    <tr>
      <th>7</th>
      <td>100002225</td>
      <td>ZDB-GENE-080219-34</td>
    </tr>
    <tr>
      <th>8</th>
      <td>100170795</td>
      <td>ZDB-GENE-080723-44</td>
    </tr>
    <tr>
      <th>9</th>
      <td>100000750</td>
      <td>ZDB-GENE-081223-2</td>
    </tr>
    <tr>
      <th>10</th>
      <td>100007521</td>
      <td>ZDB-GENE-090313-141</td>
    </tr>
    <tr>
      <th>11</th>
      <td>100149273</td>
      <td>ZDB-GENE-091117-28</td>
    </tr>
    <tr>
      <th>12</th>
      <td>100149794</td>
      <td>ZDB-GENE-110309-3</td>
    </tr>
    <tr>
      <th>13</th>
      <td>100329897</td>
      <td>ZDB-GENE-120215-92</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>You can save the data in the same way. In this case, since we have a <code class="docutils literal notranslate"><span class="pre">Pandas</span> <span class="pre">DataFrame</span></code> with column headings, the column names will be saved automatically.</p>
<p>If you would like to read the Gene IDs in from a file, all steps are the same besides the initialization of the Gene IDs:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">data_file_path</span> <span class="o">=</span> <span class="s1">&#39;data/test_data/ncbi_gene_list.txt&#39;</span>
<span class="n">list_of_gene_ids</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">data_file_path</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>

<span class="n">current_gene_id_type</span> <span class="o">=</span> <span class="s1">&#39;NCBI Gene ID&#39;</span>
<span class="n">desired_gene_id_type</span> <span class="o">=</span> <span class="s1">&#39;ZFIN ID&#39;</span>

<span class="c1"># do conversion</span>
<span class="n">converted_id_table</span> <span class="o">=</span> <span class="n">mapping</span><span class="o">.</span><span class="n">convert_ids</span><span class="p">(</span><span class="n">list_of_gene_ids</span><span class="p">,</span> <span class="n">current_gene_id_type</span><span class="p">,</span> <span class="n">desired_gene_id_type</span><span class="p">,</span> <span class="n">keep_mapping</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="n">converted_id_table</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>NCBI Gene ID</th>
      <th>ZFIN ID</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>57924</td>
      <td>ZDB-GENE-000616-10</td>
    </tr>
    <tr>
      <th>1</th>
      <td>794572</td>
      <td>ZDB-GENE-050419-85</td>
    </tr>
    <tr>
      <th>2</th>
      <td>564772</td>
      <td>ZDB-GENE-080415-1</td>
    </tr>
    <tr>
      <th>3</th>
      <td>30459</td>
      <td>ZDB-GENE-991229-12</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>The above methodologies are great if you have list of Gene IDs you would like to convert. There are cases where you might have a large dataset and one column in that dataset has the Gene IDs that you would like to convert. In this scenario, keeping all columns properly sorted is extremely important.</p>
</section>
<section id="convert-gene-ids-in-a-column-of-a-larger-dataset">
<h3>Convert Gene IDs in a Column of a Larger Dataset<a class="headerlink" href="#convert-gene-ids-in-a-column-of-a-larger-dataset" title="Permalink to this heading"></a></h3>
<p><em>Purpose: you have a dataset with columns x, y, z. Column x has Gene IDs in type A. You would like to convert these Gene IDs to type B while maintaining the information of columns y, z.</em></p>
<p>In this scenario, you might have some data that looks like:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>NCBI Gene ID</p></th>
<th class="head"><p>PValue</p></th>
<th class="head"><p>logFC</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>100002263</p></td>
<td><p>2.3</p></td>
<td><p>0.03</p></td>
</tr>
<tr class="row-odd"><td><p>…</p></td>
<td><p>…</p></td>
<td><p>…</p></td>
</tr>
</tbody>
</table>
<p>The information in the log2FC and PValue columns are essential to keep ‘in order’ with the GeneID column. It is often that in this scenario, you will have an entire gene set and will be dealing with a lot more data. Lets look at a test dataset for this case.</p>
<p><strong>Step 1:</strong> Read in the data. The data in the <code class="docutils literal notranslate"><span class="pre">test_data</span></code> sub-directory is in a <code class="docutils literal notranslate"><span class="pre">tsv</span></code> type format with a <code class="docutils literal notranslate"><span class="pre">.txt</span></code> extension. The <code class="docutils literal notranslate"><span class="pre">pandas</span></code> package can read this without an issue (same with <code class="docutils literal notranslate"><span class="pre">excel</span></code> or a <code class="docutils literal notranslate"><span class="pre">csv</span></code> file), we just need to specify the separator. <code class="docutils literal notranslate"><span class="pre">\t</span></code> is really the best for this type of data. Note that any excel file or csv file should work here.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">data_file_path</span> <span class="o">=</span> <span class="s1">&#39;data/test_data/example_diff_express_data.txt&#39;</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">data_file_path</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<p>To get a quick look at the data, we can print the first three table entries and some data stats:</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="c1"># print first three lines</span>
<span class="n">data</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>
<span class="n">rows</span><span class="p">,</span> <span class="n">cols</span> <span class="o">=</span> <span class="n">data</span><span class="o">.</span><span class="n">shape</span>
<span class="nb">print</span><span class="p">(</span><span class="sa">f</span><span class="s1">&#39;There are </span><span class="si">{</span><span class="n">rows</span><span class="si">}</span><span class="s1"> rows and </span><span class="si">{</span><span class="n">cols</span><span class="si">}</span><span class="s1"> columns&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output stream highlight-myst-ansi notranslate"><div class="highlight"><pre><span></span>There are 5464 rows and 3 columns
</pre></div>
</div>
</div>
</div>
<p><strong>Step 2:</strong> Convert Gene IDs. To execute the mapping and add the column, we will use the <code class="docutils literal notranslate"><span class="pre">add_mapped_column</span></code> function from the <code class="docutils literal notranslate"><span class="pre">mapping</span></code> module. We will give this function the data read in above, and we need to specify the <code class="docutils literal notranslate"><span class="pre">id_from</span></code> and <code class="docutils literal notranslate"><span class="pre">id_to</span></code> as before.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">id_from</span> <span class="o">=</span> <span class="s1">&#39;NCBI Gene ID&#39;</span>
<span class="n">id_to</span> <span class="o">=</span> <span class="s1">&#39;ZFIN ID&#39;</span>
<span class="n">new_data</span> <span class="o">=</span> <span class="n">mapping</span><span class="o">.</span><span class="n">add_mapped_column</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">id_from</span><span class="p">,</span> <span class="n">id_to</span><span class="p">)</span>
<span class="n">new_data</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>NCBI Gene ID</th>
      <th>ZFIN ID</th>
      <th>PValue</th>
      <th>logFC</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>100000006</td>
      <td>ZDB-GENE-030131-5654</td>
      <td>0.792615</td>
      <td>0.115009</td>
    </tr>
    <tr>
      <th>1</th>
      <td>100000044</td>
      <td>ZDB-GENE-121214-60</td>
      <td>0.015286</td>
      <td>0.803879</td>
    </tr>
    <tr>
      <th>2</th>
      <td>100000085</td>
      <td>ZDB-GENE-030131-1312</td>
      <td>0.264762</td>
      <td>0.267360</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p><strong>Step 3:</strong> Save to file. You can save this dataframe to a file as done previously.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">file_name</span> <span class="o">=</span> <span class="s1">&#39;data/out_data/converted_dataframe.txt&#39;</span>
<span class="n">utils</span><span class="o">.</span><span class="n">save_data</span><span class="p">(</span><span class="n">new_data</span><span class="p">,</span> <span class="n">file_name</span><span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<section id="special-cases">
<h4>Special Cases<a class="headerlink" href="#special-cases" title="Permalink to this heading"></a></h4>
<p>Reminder that the <code class="docutils literal notranslate"><span class="pre">id_from</span></code> and <code class="docutils literal notranslate"><span class="pre">id_to</span></code> must match the specificed options: NCBI Gene ID, Symbol, ZFIN ID, or Ensembl Gene ID. If the column in your dataset does not match one of the ID type options, that is fine. The <code class="docutils literal notranslate"><span class="pre">add_mapped_column</span></code> function has an option to deal with that.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">data_file_path</span> <span class="o">=</span> <span class="s1">&#39;data/test_data/test_set_invalid_col_name.txt&#39;</span>
<span class="n">data</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">data_file_path</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">)</span>
<span class="n">data</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>NCBI ID</th>
      <th>PValue</th>
      <th>logFC</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>100000006</td>
      <td>0.792615</td>
      <td>0.115009</td>
    </tr>
    <tr>
      <th>1</th>
      <td>100000009</td>
      <td>0.607285</td>
      <td>-0.144714</td>
    </tr>
    <tr>
      <th>2</th>
      <td>100000026</td>
      <td>0.021338</td>
      <td>0.603871</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>As you can see here, the column name we have is <code class="docutils literal notranslate"><span class="pre">NCBI</span> <span class="pre">ID</span></code>, which is not one of the options. We can use the <code class="docutils literal notranslate"><span class="pre">column_name_with_ids</span></code> parameter to get around this. Note that the <code class="docutils literal notranslate"><span class="pre">id_from</span></code> parameter must be one of the options and the <code class="docutils literal notranslate"><span class="pre">column_name_with_ids</span></code> must match the column name with the IDs in the dataset exactly. Of course, we would want <code class="docutils literal notranslate"><span class="pre">id_from</span></code> and <code class="docutils literal notranslate"><span class="pre">column_name_with_ids</span></code> to be of the same Gene ID type, likely with just a different spelling or naming convention as seen here.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">id_from</span> <span class="o">=</span> <span class="s1">&#39;NCBI Gene ID&#39;</span>
<span class="n">id_to</span> <span class="o">=</span> <span class="s1">&#39;ZFIN ID&#39;</span>
<span class="n">column_name_with_ids</span> <span class="o">=</span> <span class="s1">&#39;NCBI ID&#39;</span>
<span class="n">new_data</span> <span class="o">=</span> <span class="n">mapping</span><span class="o">.</span><span class="n">add_mapped_column</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">id_from</span><span class="p">,</span> <span class="n">id_to</span><span class="p">,</span> <span class="n">column_name_with_ids</span><span class="p">)</span>
<span class="n">new_data</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>NCBI ID</th>
      <th>ZFIN ID</th>
      <th>PValue</th>
      <th>logFC</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>100000006</td>
      <td>ZDB-GENE-030131-5654</td>
      <td>0.792615</td>
      <td>0.115009</td>
    </tr>
    <tr>
      <th>1</th>
      <td>100000009</td>
      <td>ZDB-GENE-130530-778</td>
      <td>0.607285</td>
      <td>-0.144714</td>
    </tr>
    <tr>
      <th>2</th>
      <td>100000026</td>
      <td>ZDB-GENE-120823-1</td>
      <td>0.021338</td>
      <td>0.603871</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>We can see that the mapping was executed successfully even though the column name is not one of our specified options.</p>
<p>In the final special case, we could also drop the old column ids if we don’t need them anymore.</p>
<div class="cell docutils container">
<div class="cell_input docutils container">
<div class="highlight-ipython3 notranslate"><div class="highlight"><pre><span></span><span class="n">new_data</span> <span class="o">=</span> <span class="n">mapping</span><span class="o">.</span><span class="n">add_mapped_column</span><span class="p">(</span><span class="n">data</span><span class="p">,</span> <span class="n">id_from</span><span class="p">,</span> <span class="n">id_to</span><span class="p">,</span> 
                                     <span class="n">column_name_with_ids</span><span class="p">,</span> <span class="n">keep_old_ids</span><span class="o">=</span><span class="kc">False</span><span class="p">)</span>
<span class="n">new_data</span><span class="o">.</span><span class="n">head</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="cell_output docutils container">
<div class="output text_html"><div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>ZFIN ID</th>
      <th>PValue</th>
      <th>logFC</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>ZDB-GENE-030131-5654</td>
      <td>0.792615</td>
      <td>0.115009</td>
    </tr>
    <tr>
      <th>1</th>
      <td>ZDB-GENE-130530-778</td>
      <td>0.607285</td>
      <td>-0.144714</td>
    </tr>
    <tr>
      <th>2</th>
      <td>ZDB-GENE-120823-1</td>
      <td>0.021338</td>
      <td>0.603871</td>
    </tr>
  </tbody>
</table>
</div></div></div>
</div>
<p>In this run I specified the parameter <code class="docutils literal notranslate"><span class="pre">keep_old_ids</span></code> to be <code class="docutils literal notranslate"><span class="pre">False</span></code>; therefore, it is no longer included in the output.</p>
</section>
</section>
</section>
<section id="conclusion">
<h2>Conclusion<a class="headerlink" href="#conclusion" title="Permalink to this heading"></a></h2>
<p>This concludes the mapping tutorial! In summary, the key functions in this library for mapping are:</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>function</p></th>
<th class="head"><p>purpose</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>convert_ids</p></td>
<td><p>convert a list of Gene IDs</p></td>
</tr>
<tr class="row-odd"><td><p>add_mapped_column</p></td>
<td><p>add a converted list of Gene IDs to an existing DataFrame</p></td>
</tr>
</tbody>
</table>
<p>For more information about the full functionality of each function, please refer to the API Reference.</p>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="../autoapi/danrerlib/mapping/index.html" class="btn btn-neutral float-left" title="danrerlib.mapping" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
        <a href="perform_orthology_mapping.html" class="btn btn-neutral float-right" title="Orthology Mapping" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2023, Ashley Schwartz.</p>
  </div>

   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>